AirCasting/SessionsSynchronization/Controller/SessionSynchronizationController.swift

```diff
@@ -6,18 +6,20 @@ import Combine
 import CoreLocation
 
 final class SessionSynchronizationController: SessionSynchronizer {
+    
     /// A plugin point for error handlers
     var errorStream: SessionSynchronizerErrorStream?
     
     private let synchronizationContextProvider: SessionSynchronizationContextProvidable
     private let downstream: SessionDownstream
     private let upstream: SessionUpstream
     private let store: SessionSynchronizationStore
-    private let dataConverter = SynchronizationDataConterter()
+    private let dataConverter = SynchronizationDataConverter()
+    private let viewModel: SessionSynchronizationViewModel
     
     // Progress tracking for filtering requests while already syncing
     // (can this be somehow moved to a custom operator or something?)
-    private var syncInProgress: Bool = false
+//    var syncInProgress: Bool = false
```

Comment:
this flag and how it's used in the `triggerSynchronization(completion:)` function is **super crucial** for avoiding race conditions, it cannot be removed.

Basically this class should not be modified in order to accomplish the task. I would probably use a [decorator](https://refactoring.guru/pl/design-patterns/decorator) pattern in order to add something like a published bool to indicate if synchronization is in progress. The other way is to ever so slightly change this just by making this `syncInProgress` public and also `@Published` and make our way from there. 
The dependency needs to be the exact oposite as here - `ViewModel` should hold a `SessionSynchronizer`, not the other way around (dependency inversion principle, also single responsibility principle)